.github/workflows/check-out-of-tree-build.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow is intended to check if PR doesn't introduce any regressions
+# into out-of-tree build in various configurations.
+#
+# Documentation for GitHub Actions:
+# [workflow-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+# [context-and-expression-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions
+name: Out-of-tree build & tests
+
+on:
+  push:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check build for documentation changes
+      - 'docs/**'
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check build for documentation changes
+      - 'docs/**'
+      - '**.md'
+  schedule:
+    - cron: 0 0 * * *
+
+env:
+  LLVM_VERSION: 12
+
+jobs:
+  build_and_test:
+    name: Linux
+    strategy:
+      matrix:
+        build_type: [Release, Debug]
```

Comment:
Do we link against static or shared LLVM libraries here? In Travis CI we had both options. Should we have the same behavior in GH Actions or having just one (shared libs for example) is enough for us?

Reply:
> Do we link against static or shared LLVM libraries here? In Travis CI we had both options. 

Let me take a more detailed look into this: the exact modes which are available for us in out-of-tree configuration are actually limited by LLVM distribution, i.e. if there were no shared libraries in it, we won't be able to test such setup. But I guess we have everything in place if we had both configurations in Travis CI

Reply:
> Let me take a more detailed look into this

But I suggest that we merge this earlier and enable more configurations later as it might take a few days for me to find time to look into this